procrustes/psdp.py

```diff
@@ -549,3 +675,29 @@ def _scale(e: np.ndarray, g: np.ndarray, q: np.ndarray) -> np.ndarray:
         )
     )
     return alpha * e
+
+
+def _psd_proj(arr: np.ndarray, do_cholesky: bool = True) -> np.ndarray:
+    r"""
+    Return the symmetric positive semi-definite matrix nearest to a given matrix.
+
+    Parameters
+    ----------
+    arr : np.ndarray
+        The input matrix.
+
+    Returns
+    -------
+    np.ndarray
+        The nearest symmetric positive semi-definite matrix.
+    """
+    arr = (arr + arr.T) / 2
+    if np.isnan(arr).any() or np.isposinf(abs(arr)).any():
+        raise ValueError("C^h u.T")
```

Comment:
The error information is not clear. I think we should mention that there are `NaN` or infinity values. *f-string* might be helpful.